core/src/main/scala/fs2/concurrent.scala

```diff
@@ -23,8 +44,10 @@ object concurrent {
               }
               else F.pure(())
             }}}
-          s.chunks.attempt.evalMap { o => outputQueue.enqueue1(Some(o)) }
-           .interruptWhen(killSignal).run.flatMap { _ => Free.eval(done) }.run
+          inner.chunks.attempt.evalMap { o => outputQueue.enqueue1(Some(o)) }
+          .interruptWhen(killSignal)
+          .onFinalize(done)
+          .run.run
         }}
```

Comment:
I don't actually see how this change could make a difference... Are you sure it does? Because of the attempt there does not seem to be any way that the run can fail.